src/inc/til/string.h

```diff
@@ -373,4 +374,20 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
             return compare_string_ordinal(lhs, rhs) == CSTR_LESS_THAN;
         }
     };
+
+    // Implement to_int in terms of to_ulong by negating its result. to_ulong does not expect
+    // to be passed signed numbers and will return an error accordingly. That error when
+    // compared against -1 evaluates to true. We account for that by returning to_int_error if to_ulong
+    // returns an error.
+    constexpr int to_int(const std::wstring_view& str, unsigned long base = 0) noexcept
+    {
+        const auto signPosition = str.find(L"-");
+        if (signPosition != std::wstring_view::npos)
+        {
+            const auto result = to_ulong(str.substr(signPosition + 1), base);
+            return result == to_ulong_error ? to_int_error : result * -1;
+        }
+
+        return to_ulong(str, base) == to_ulong_error ? to_int_error : to_ulong(str, base);

```

Comment:
Awesome! Thanks for doing this.

One remaining concern is that I think we should return `to_int_error` if the resultant value would be outside of the range `INT_MIN .. INT_MAX`

Reply:
I'll work on that tonight. Today was rough at work 😫

Reply:
aw, I'm sorry!

Reply:
Pushed a change